File: 637. Average of Levels in Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
  def averageOfLevels(self, root):
    """
    :type root: TreeNode
    :rtype: List[float]
    """
    def bfs(start):
      myQueue = [start]
      result = []
      while myQueue:
        total = 0
        count = 0
        newQueue = []
        for index in range(len(myQueue)):
          current = myQueue[index]
          if current != None:
            total += current.val
            count += 1
          if current.left != None:
            newQueue.append(current.left)
          if current.right != None:
            newQueue.append(current.right)
        average = float(total) / float(count)
        result.append(average)
        myQueue = newQueue
      return result
    
    return bfs(root)

File: 637. Average of Levels in Binary Tree/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var averageOfLevels = function(root) {
  function bfs(start){
    let myQueue = [start];
    let result = [];
    while (myQueue.length != 0){
      let total = 0;
      let count = 0;
      let newQueue = [];
      for (let index = 0; index < myQueue.length; index++){
        let current = myQueue[index];
        if (current != undefined){
          total += current.val;
          count += 1;
        }
        if (current.left != undefined){
          newQueue.push(current.left);
        }
        if (current.right != undefined){
          newQueue.push(current.right);
        }
      }
      let average = total / count;
      result.push(average);
      myQueue = newQueue;
    }
    return result;
  }
  
  return bfs(root);
};